src/System.Security.Cryptography.X509Certificates/tests/CollectionTests.cs

```diff
@@ -360,6 +376,136 @@ public static void X509CertificateCollectionSyncRoot()
+                // but the order isn't really a factor.
+                Assert.Equal(collection.Count, importedCollection.Count);
+
+                // Compare just the subject names first, because it's the easiest thing to read out of the failure message.
+                string[] subjects = new string[collection.Count];
+                string[] importedSubjects = new string[collection.Count];
+
+                for (int i = 0; i < collection.Count; i++)
+                {
+                    subjects[i] = collection[i].GetNameInfo(X509NameType.SimpleName, false);
+                    importedSubjects[i] = importedCollection[i].GetNameInfo(X509NameType.SimpleName, false);
+                }
+
+                Assert.Equal(subjects, importedSubjects);
+
+                // But, really, the collections should be equivalent
+                // (after being coerced to IEnumerable<X509Certificate2>)
+                Assert.Equal(collection.OfType<X509Certificate2>(), importedCollection.OfType<X509Certificate2>());
+            }
+        }
+
+        [Fact]
+        public static void MultipleImport()
+        {
+            var collection = new X509Certificate2Collection();
+
+            collection.Import(Path.Combine("TestData", "DummyTcpServer.pfx"), null, default(X509KeyStorageFlags));
+            collection.Import(TestData.PfxData, TestData.PfxDataPassword, default(X509KeyStorageFlags));
+
+            Assert.Equal(3, collection.Count);
+        }
+
+        [Fact]
+        [ActiveIssue(2743, PlatformID.AnyUnix)]
+        public static void ExportMultiplePrivateKeys()
+        {
+            var collection = new X509Certificate2Collection();
+
+            collection.Import(Path.Combine("TestData", "DummyTcpServer.pfx"), null, X509KeyStorageFlags.Exportable);
+            collection.Import(TestData.PfxData, TestData.PfxDataPassword, X509KeyStorageFlags.Exportable);
+
+            // Pre-condition, we have multiple private keys
+            int originalPrivateKeyCount = collection.OfType<X509Certificate2>().Count(c => c.HasPrivateKey);
+            Assert.Equal(2, originalPrivateKeyCount);
+
+            // Export, re-import.
+            byte[] exported;
+
+            try
+            {
+                exported = collection.Export(X509ContentType.Pkcs12);
+            }
+            catch (PlatformNotSupportedException)
+            {
+                // [ActiveIssue(2743, PlatformID.AnyUnix)]
+                // Our Unix builds can't export more than one private key in a single PFX, so this is
+                // their exit point.
+                //
+                // If Windows gets here, or any exception other than PlatformNotSupportedException is raised,
+                // let that fail the test.
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    throw;
+                }
+
+                return;
+            }
+
```

Comment:
Do we need a check here that RuntimeInformation.IsOSPlatform(OSPlatform.Windows) is true, to verify that we did get a PNSE on Unix?